Bindings/Packages/com.virtualmaker.bindings/Runtime/Bindings.InputActions.cs

```diff
@@ -0,0 +1,109 @@
+            where T : struct
+        {
+            var inputAction = inputActionReference.GetAction();
+            inputAction.Enable();
+            inputAction.performed += OnActionPerformed;
+
+            AddUnsubscriber(() =>
+            {
+                inputAction.Disable();
+                inputAction.performed -= OnActionPerformed;
+            });
+
+            if (started != null)
+            {
+                inputAction.started += OnActionStarted;
+                AddUnsubscriber(() => inputAction.started -= OnActionStarted);
+            }
+
+            if (cancelled != null)
+            {
+                inputAction.canceled += OnActionCanceled;
+                AddUnsubscriber(() => inputAction.canceled -= OnActionCanceled);
+            }
+
+            return;
+            void OnActionPerformed(InputAction.CallbackContext ctx) => performed(ctx.ReadValue<T>());
+            void OnActionStarted(InputAction.CallbackContext ctx) => started(ctx.ReadValue<T>());
+            void OnActionCanceled(InputAction.CallbackContext ctx) => cancelled(ctx.ReadValue<T>());
+        }
+
+        public void Bind(InputActionReference inputActionReference, Action<InputAction.CallbackContext> performed, Action<InputAction.CallbackContext> started = null, Action<InputAction.CallbackContext> cancelled = null)
+        {
+            var inputAction = inputActionReference.GetAction();
+            inputAction.Enable();
+            inputAction.performed += OnActionPerformed;
+
+            AddUnsubscriber(() =>
+            {
+                inputAction.Disable();
+                inputAction.performed -= OnActionPerformed;
+            });
+
+            if (started != null)
+            {
+                inputAction.started += OnActionStarted;
+                AddUnsubscriber(() => inputAction.started -= OnActionStarted);
+            }
+
+            if (cancelled != null)
+            {
+                inputAction.canceled += OnActionCanceled;
+                AddUnsubscriber(() => inputAction.canceled -= OnActionCanceled);
+            }
+
+            return;
+            void OnActionPerformed(InputAction.CallbackContext ctx) => performed(ctx);
+            void OnActionStarted(InputAction.CallbackContext ctx) => started(ctx);
+            void OnActionCanceled(InputAction.CallbackContext ctx) => cancelled(ctx);
+        }
+    }
+
+    internal static class InputSystemExtensions
+    {
+        public static InputAction GetAction(this InputActionReference reference)
+        {
+            var action = InputSystem.actions.FindAction(reference.action.id);
+
+            if (action == null)
+            {
+                throw new MissingReferenceException($"Failed to find input action for {reference.name}");
+            }
+
+            return action;
+        }
+    }
+}
+#endif // UNITY_INPUT_SYSTEM
```

Comment:
```suggestion

#endif // UNITY_INPUT_SYSTEM
```

Reply:
I actually prefer that there wasn't a space here.